### ISSUE TITLE
Syntactic sugar for `target_compatible_with` a `bool_flag`, `config_setting`

### ISSUE THREAD

**Issue by tpudlik** [bazelbuild/bazel#21857]
### Description of the feature request:

I would like to mark a target as `target_compatible_with` a `config_setting`. I hoped I could do,

```
target_compatible_with = [":my_config_setting"]
```

It turns out this is not supported. Instead, I must express this as,

```
target_compatible_with = select({
  ":my_config_setting": [],
  "//conditions:default": ["@platforms//:incompatible"],
})
```

That's a little verbose, and a little obscure!

Actually, things are even worse than this, because the `config_setting` is used only to detect that a Skylib `bool_flag` is true. What I'd really love to say is,

```
bool_flag(
  name = "my_flag",
  build_setting_default = False,
)

cc_library(
  target_compatible_with = [":my_flag"],
)
```

But instead I write,

```
bool_flag(
  name = "my_flag",
  build_setting_default = False,
)

config_setting(
  name = "my_flag_true",
  flag_values =  {
    ":my_flag": "True",
  },
)

cc_library(
  target_compatible_with = select({
    ":my_flag_true": [],
    "//conditions:default": ["@platforms//:incompatible"],
  }),
)
```

@katre

### Which category does this issue belong to?

Configurability

**Comment by katre** [comment]
To clarify, the underlying problem is that:
1. `target_compatible_with` expects to receive `constraint_value` targets
2. You are instead passing it a `config_setting`, and
3. The special value `@platforms//:incompatible` isn't actually special to bazel at all, it's just defined in a way that no platform will ever match it.

We can certainly take a look at supporting this, but I don't actually think it's going to be a clean change to the current support.

**Comment by tpudlik** [comment]
Yeah, that's right. I just wanted to report that this user journey is a bit awkward for me. But I understand it may not be easy to smooth out!

**Comment by keith** [comment]
https://github.com/bazelbuild/bazel/pull/29475

**Comment by fmeum** [comment]
@bazel-io fork 9.2.0

**Comment by iancha1992** [comment]
This feature has been included in [Bazel 9.2.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/9.2.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=9.2.0rc1`. Thanks!
